src/routes/chains/openapi/api-chain-page.ts

```diff
@@ -0,0 +1,16 @@
+import { Page as DomainPage } from '../../../common/entities/page.entity';
+import { ApiProperty } from '@nestjs/swagger';
+import { Chain as ApiChain } from './api-chain';
+
+export class Page implements DomainPage<ApiChain> {
+  @ApiProperty()
+  count: number;
+  @ApiProperty()
+  next: string;
+  @ApiProperty()
+  previous: string;
+  @ApiProperty({
+    type: ApiChain,
+  })
+  results: ApiChain[];
```

Comment:
In order to correctly generate the stubs for a `Page` (which is polymorphic) we need to specify the types that can be part of `Page`.

Since we are taking a modular approach I've created a `Page` just for `Chain`. The alternative would be to have a `Page` that would set here all possible types that we expose with `Page`.

This will result in more repeated classes but I see these classes purely for documentation purposes (ie.: they are not used outside of our routes).